packages/collaboration-extension/src/filebrowser.ts

```diff
@@ -160,6 +163,92 @@ export const defaultFileBrowser: JupyterFrontEndPlugin<IDefaultFileBrowser> = {
   }
 };
 
+/**
+ * The default collaborative drive provider.
+ */
+export const logger: JupyterFrontEndPlugin<void> = {
+  id: '@jupyter/collaboration-extension:logger',
+  description: 'A logging plugin for debugging purposes.',
+  autoStart: true,
+  optional: [ILoggerRegistry, INotebookTracker, ITranslator],
+  activate: (
+    app: JupyterFrontEnd,
+    loggerRegistry: ILoggerRegistry | null,
+    nbtracker: INotebookTracker | null,
+    translator: ITranslator | null
+  ): void => {
+    const trans = (translator ?? nullTranslator).load('jupyter_collaboration');
+    const schemaID =
+      'https://events.jupyter.org/jupyter_server/jupyter_collaboration/v1';
+
+    if (!loggerRegistry) {
+      app.serviceManager.events.stream.connect((_, emission) => {
+        if (emission.schema_id === schemaID) {
+          console.debug(
+            `[${emission.room}(${emission.path})] ${emission.action ?? ''}: ${
+              emission.msg ?? ''
+            }`
+          );
+
+          if (emission.level === 'WARNING') {
+            showDialog({
+              title: trans.__('Warning'),
+              body: trans.__(
+                `Two collaborative sessions are accessing the file ${emission.path} simultaneously.
+                \nOpening the same file using different views simultaneously is not supported. Please, close one view; otherwise, you might lose some of your progress.`
+              ),
+              buttons: [Dialog.okButton()]
+            });
+          }
+        }
+      });
+
+      return;
+    }
+
+    const loggers: Map<string, ILogger> = new Map();
+
+    if (nbtracker) {
+      nbtracker.widgetAdded.connect((sender, nb) => {
+        const logger = loggerRegistry.getLogger(nb.context.path);
+        loggers.set(nb.context.localPath, logger);
+
+        nb.disposed.connect(nb => {
+          loggers.delete(nb.context.localPath);
+        });
+      });
+    }
+
+    void (async () => {
+      const { events } = app.serviceManager;
+      for await (const emission of events.stream) {
+        if (emission.schema_id === schemaID) {
+          const logger = loggers.get(emission.path as string);
+
+          logger?.log({
+            type: 'text',
+            level: (emission.level as string).toLowerCase() as any,
+            data: `[${emission.room}] ${emission.action ?? ''}: ${
+              emission.msg ?? ''
+            }`
+          });
+
+          if (emission.level === 'WARNING') {
+            showDialog({
+              title: trans.__('Warning'),
+              body: trans.__(
+                `Two collaborative sessions are accessing the file ${emission.path} simultaneously.
+                \nOpening the same file using different views simultaneously is not supported. Please, close one view; otherwise, you might lose some of your progress.`
+              ),
+              buttons: [Dialog.okButton()]
```

Comment:
What about using the _red_ warning button?

```suggestion
              buttons: [Dialog.warnButton({ label: trans.__('Ok') })]
```